libs/util/Auxilary.arucas

```diff
@@ -69,25 +87,37 @@ class VarArgs {
 
     fun get(index: Number): Object {
         if (index >= this.size) {
-            throw new Error("Index out of bounds");
+            throw new IndexError(index, this.size);
         }
         return this.arguments.get(i);
     }
 
     fun get(index: Number, type: Type): Object {
         value = this.get(index);
         if (!value.instanceOf(type)) {
-            throw new Error("Value was of an invalid type");
+            throw new TypeError(type, Type.of(value));
         }
         return value;
     }
 
-    fun getOrDefault(index: Number, type: Type, defaultValue: Object): Object {
-        value = this.get(index);
-        if (value.instanceOf(type)) {
-            return value;
+    fun getOrDefault(index: Number, defaultValue: Object): Object {
+        value = defaultValue;
+        try {
+            value = this.get(index);
+        }
+        catch (error) {
+            if (error.instanceOf(IndexError.type)) return value;
```

Comment:
This is kinda weird. Why not just check the size, and if it's out of bounds then return default instead of doing this weird catch check.

Reply:
yup in this case no other error is thrown except index error, but I feel like there should be proper way to catch certain error types